pkg/apis/eksctl.io/v1alpha5/validation_test.go

```diff
@@ -5,129 +5,169 @@ import (
+			})
 
-			err = validateInstancesDistribution(ng)
-			Expect(err).ToNot(HaveOccurred())
-		})
+			It("It fails when the onDemandBaseCapacity is not above 0", func() {
+				ng.InstancesDistribution.OnDemandBaseCapacity = newInt(-1)
 
-		It("It fails when the spotInstancePools is not between 1 and 20", func() {
-			ng.InstancesDistribution.SpotInstancePools = newInt(0)
+				err := validateInstancesDistribution(ng)
+				Expect(err).To(HaveOccurred())
 
-			err := validateInstancesDistribution(ng)
-			Expect(err).To(HaveOccurred())
+				ng.InstancesDistribution.OnDemandBaseCapacity = newInt(1)
 
-			ng.InstancesDistribution.SpotInstancePools = newInt(21)
-			err = validateInstancesDistribution(ng)
-			Expect(err).To(HaveOccurred())
+				err = validateInstancesDistribution(ng)
+				Expect(err).ToNot(HaveOccurred())
+			})
 
-			ng.InstancesDistribution.SpotInstancePools = newInt(2)
-			err = validateInstancesDistribution(ng)
-			Expect(err).ToNot(HaveOccurred())
-		})
+			It("It fails when the spotInstancePools is not between 1 and 20", func() {
+				ng.InstancesDistribution.SpotInstancePools = newInt(0)
 
-		It("It fails when the onDemandPercentageAboveBaseCapacity is not between 0 and 100", func() {
-			ng.InstancesDistribution.OnDemandPercentageAboveBaseCapacity = newInt(-1)
+				err := validateInstancesDistribution(ng)
+				Expect(err).To(HaveOccurred())
 
-			err := validateInstancesDistribution(ng)
-			Expect(err).To(HaveOccurred())
+				ng.InstancesDistribution.SpotInstancePools = newInt(21)
+				err = validateInstancesDistribution(ng)
+				Expect(err).To(HaveOccurred())
 
-			ng.InstancesDistribution.OnDemandPercentageAboveBaseCapacity = newInt(101)
-			err = validateInstancesDistribution(ng)
-			Expect(err).To(HaveOccurred())
+				ng.InstancesDistribution.SpotInstancePools = newInt(2)
+				err = validateInstancesDistribution(ng)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It("It fails when the onDemandPercentageAboveBaseCapacity is not between 0 and 100", func() {
+				ng.InstancesDistribution.OnDemandPercentageAboveBaseCapacity = newInt(-1)
+
+				err := validateInstancesDistribution(ng)
+				Expect(err).To(HaveOccurred())
+
+				ng.InstancesDistribution.OnDemandPercentageAboveBaseCapacity = newInt(101)
+				err = validateInstancesDistribution(ng)
+				Expect(err).To(HaveOccurred())
+
+				ng.InstancesDistribution.OnDemandPercentageAboveBaseCapacity = newInt(50)
+				err = validateInstancesDistribution(ng)
+				Expect(err).ToNot(HaveOccurred())
+			})
+		})
+	})
+
+	Describe("kubelet extra config", func() {
+		Context("Instances distribution", func() {
+
+			var ng *NodeGroup
+			BeforeEach(func() {
+				ng = &NodeGroup{}
+			})
+
+			It("Forbids overriding basic fields", func() {
+				testKeys := []string{"kind", "apiVersion", "address", "clusterDomain", "authentication",
+					"authorization", "serverTLSBootstrap", "featureGates"}
+
+				for _, key := range testKeys {
+					ng.KubeletExtraConfig = &NodeGroupKubeletConfig{
```

Comment:
Nice! For some reason I thought you always have to use `map[string]insterface{}` to construct the thing...